### PR TITLE
Sync comments with dokan-dotnet and done some Doxygen fix

### PR DESCRIPTION
**Pull request by magol**
I hope that I have done the work with GIT correct.

### CODE REVIEW

File: dokan/dokan.h [hunk 1]
```diff
@@ -69,7 +69,7 @@ extern "C" {
 
 /**
  * \defgroup DOKAN_OPTION DOKAN_OPTION
- * \brief All DOKAN_OPTION flags used in DOKAN_OPTIONS.Options
+ * \brief All DOKAN_OPTION flags used in \ref DOKAN_OPTIONS.Options
```
Comment (Liryna) [inline dokan/dokan.h]: For information, doxygen does not always need the \ref. In this case, doxygen already understood it without \ref.

File: dokan/dokan.h [hunk 2]
```diff
@@ -136,23 +137,23 @@ typedef struct _DOKAN_FILE_INFO {
...
   /**
-   * Process id for the thread that originally requested a given I/O operation
+   * Process id for the thread that originally requested a given I/O operation.
    */
   ULONG ProcessId;
   /**
-   * Requesting a directory file
-   * IsDirectory has to be set in CreateFile is the file appear to be a folder.
+   * Requesting a directory file.
+   * Must be set in \ref DOKAN_OPERATIONS#ZwCreateFile if the file appear to be a folder.
```
Comment (Liryna) [inline dokan/dokan.h]: There is a difference with DOKAN_OPERATIONS#ZwCreateFile and DOKAN_OPERATIONS.ZwCreateFile ?
Both seems to work :D

File: dokan/dokan.h [hunk 3]
```diff
@@ -136,23 +137,23 @@ typedef struct _DOKAN_FILE_INFO {
...
    */
   UCHAR IsDirectory;
-  /** Flag if the file has to be delete during DOKAN_OPERATIONS.Cleanup event. */
+  /** Flag if the file has to be delete during \ref DOKAN_OPERATIONS#Cleanup event. */
```
Comment (Liryna) [inline dokan/dokan.h]: Same as https://github.com/dokan-dev/dokany/pull/314/files#r74699351

File: dokan/dokan.h [hunk 4]
```diff
@@ -243,6 +243,9 @@ typedef struct _DOKAN_OPERATIONS {
   * \brief CloseFile Dokan API callback
   *
   * Clean remaining Context
+  * 
+  * CloseFile is called at the end of the life of the context.
+  * Remainings in \ref DOKAN_FILE_INFO#Context has to be cleared before return.
```
Comment (Liryna) [inline dokan/dokan.h]: Same as https://github.com/dokan-dev/dokany/pull/314/files#r74699351

File: dokan/dokan.h [hunk 5]
```diff
@@ -253,7 +256,7 @@ typedef struct _DOKAN_OPERATIONS {
   /**
   * \brief ReadFile Dokan API callback
   *
-  * ReadFile callback on the file previously opened in CreateFile
+  * ReadFile callback on the file previously opened in \ref CreateFile.
```
Comment (Liryna) [inline dokan/dokan.h]: Should be ZwCreateFile and not CreateFile

File: dokan/dokan.h [hunk 6]
```diff
@@ -275,7 +279,7 @@ typedef struct _DOKAN_OPERATIONS {
   /**
   * \brief WriteFile Dokan API callback
   *
-  * WriteFile callback on the file previously opened in CreateFile
+  * WriteFile callback on the file previously opened in \ref CreateFile
```
Comment (Liryna) [inline dokan/dokan.h]: Should be ZwCreateFile and not CreateFile

File: dokan/dokan.h [hunk 7]
```diff
@@ -387,38 +394,46 @@ typedef struct _DOKAN_OPERATIONS {
...
+  * Check if it is possible to delete a file.
+  * 
+  * You should not delete the file in DeleteFile, but instead
   * you must only check whether you can delete the file or not,
   * and return \c STATUS_SUCCESS (when you can delete it) or appropriate error
   * codes such as \c STATUS_ACCESS_DENIED or \c STATUS_OBJECT_NAME_NOT_FOUND.
   *
   * When you return \c STATUS_SUCCESS, you get a Cleanup call afterwards with
-  * FileInfo->DeleteOnClose set to \c TRUE and only then you have to actually
+  * \ref DOKAN_FILE_INFO#DeleteOnClose set to \c TRUE and only then you have to actually
```
Comment (Liryna) [inline dokan/dokan.h]: Same as https://github.com/dokan-dev/dokany/pull/314/files#r74699351

File: dokan/dokan.h [hunk 8]
```diff
@@ -387,38 +394,46 @@ typedef struct _DOKAN_OPERATIONS {
...
+  * 
+  * You should not delete the Directory in DeleteDirectory, but instead
   * you must only check whether you can delete the file or not,
   * and return \c STATUS_SUCCESS (when you can delete it) or appropriate error
   * codes such as \c STATUS_ACCESS_DENIED, \c STATUS_OBJECT_PATH_NOT_FOUND,
   * or \c STATUS_DIRECTORY_NOT_EMPTY.
   *
   * When you return \c STATUS_SUCCESS, you get a Cleanup call afterwards with
-  * FileInfo->DeleteOnClose set to \c TRUE and only then you have to actually
+  * \ref DOKAN_FILE_INFO#DeleteOnClose set to \c TRUE and only then you have to actually
```
Comment (Liryna) [inline dokan/dokan.h]: Same as https://github.com/dokan-dev/dokany/pull/314/files#r74699351

File: dokan/dokan.h [hunk 9]
```diff
@@ -527,14 +545,12 @@ typedef struct _DOKAN_OPERATIONS {
...
-  *
-  * Neither GetDiskFreeSpace nor GetVolumeInformation
-  * save the DokanFileContext->Context.
-  *
-  * Before these methods are called, CreateFile may not be called.
-  * (ditto CloseFile and Cleanup)
+  * Retrieves information about the file system and volume associated with the specified root directory.
+  * 
+  * Neither GetVolumeInformation nor GetDiskFreeSpace
+  * save the \ref DOKAN_FILE_INFO#Context.
```
Comment (Liryna) [inline dokan/dokan.h]: Same as https://github.com/dokan-dev/dokany/pull/314/files#r74699351

File: dokan/dokan.h [hunk 10]
```diff
@@ -610,20 +627,17 @@ typedef struct _DOKAN_OPERATIONS {
...
-  * Suported since 0.6.0. You must specify the version at
-  * DOKAN_OPTIONS.Version.
+  * Sets the security of a file or directory object.
   *
+  * \since Supported since version 0.6.0. You must specify the version in \ref DOKAN_OPTIONS#Version.
   * \param FileName File path requested by the Kernel on the FileSystem.
-  * \param SecurityInformation Structure that identifies the contents of the security descriptor pointed by SecurityDescriptor param.
-  * \param SecurityDescriptor A pointer to a SECURITY_DESCRIPTOR structure.
+  * \param SecurityInformation Structure that identifies the contents of the security descriptor pointed by \a SecurityDescriptor param.
+  * \param SecurityDescriptor A pointer to a \ref SECURITY_DESCRIPTOR structure.
```
Comment (Liryna) [inline dokan/dokan.h]: No ref possible here. SECURITY_DESCRIPTOR is not define in dokan but windows headers.

File: dokan/dokan.h [hunk 11]
```diff
@@ -636,12 +650,10 @@ typedef struct _DOKAN_OPERATIONS {
...
   * This is only called if \ref DOKAN_OPTION_ALT_STREAM is enabled.
-  *
-  * Supported since 0.8.0. You must specify the version at
-  * DOKAN_OPTIONS.Version.
-  *
+  * 
+  * \since Supported since version 0.8.0. You must specify the version in \ref DOKAN_OPTIONS#Version.
   * \param FileName File path requested by the Kernel on the FileSystem.
-  * \param FillFindStreamData Callback that has to be called with PWIN32_FIND_STREAM_DATA that contain stream information.
+  * \param FillFindStreamData Callback that has to be called with \ref PWIN32_FIND_STREAM_DATA that contain stream information.
```
Comment (Liryna) [inline dokan/dokan.h]: No ref possible here. PWIN32_FIND_STREAM_DATA is not define in dokan but windows headers.

**Comment by Liryna** [comment]
You had to pull the changes before create your commit to skip the "auto merge commit".
Other this sync from dokan-dotnet is well appreciate ^^

**Comment by magol** [comment]
"You had to pull the changes before create your commit to skip the "auto merge commit"."
I do not quite understand what you mean by that, and that's probably the problem. Please, help me understand what I should do to make the right :-)

**Comment by magol** [comment]
Ok, pull is something I have to do BEFORE commit. So it is to late to do it now?

**Comment by Liryna** [comment]
Since your commit was made after the remote commit, he need to merge your commit together and sometimes you can have conflicts because both commit changed the same file. 

Normally you can fix it with `git rebase -i HEAD~1` (since you made only one commit)

**Pull request closed without merging**